### DISCUSSION TITLE
Best workflow for SDXL Hires Fix

### DISCUSSION THREAD

**Discussion by toyssamurai** [Comfy-Org/ComfyUI#1002] [Q&A] [answered]
I wonder if I have been doing it wrong -- right now, when I do latent upscaling with SDXL, I add an Upscale Latent node after the refiner's KSampler node, and pass the result of the latent upscaler to another KSampler. After that, it goes to a VAE Decode and then to a Save Image node. 

It's doing a fine job, but I am not sure if this is the best way of doing latent upscaling with SDXL. Should I put the latent upscaler before the refiner and after the base model instead? I tried it once, and the result isn't quite as good, but may be I did it wrong?

**Comment by ntdviet** [comment, marked as answer]
I studied the manipulation of latent images with leftover noise (its in your case right after the base model sampler) and surprisingly, you can not. The upscaling distort the gaussian noise from circle forms to squares and this totally ruin the next sampling step... I don't know if there is any other upscaler node that works, but the basic upscale methods all dont' do the job (nearest exact, bilinear, area, bicubic, bislerp)

So yes to your question, up to this moment, all I can see working is putting the upscaler node right after the refiner sampler, when the leftover noise is cleared and the latent is kind of "closed".
